core/trino-main/src/main/java/io/trino/server/remotetask/HttpRemoteTask.java

```diff
@@ -998,6 +998,32 @@ public void onFailure(Throwable t)
                     errorScheduledExecutor.schedule(() -> doScheduleAsyncCleanupRequest(cleanupBackoff, request, action), delayNanos, NANOSECONDS);
                 }
             }
+
+            private void fatalAsyncCleanupFailure(Throwable cause)
+            {
+                synchronized (HttpRemoteTask.this) {
+                    try (SetThreadName ignored = new SetThreadName("HttpRemoteTask-%s", taskId)) {
+                        TaskStatus taskStatus = getTaskStatus();
+
+                        if (taskStatus.getState().isDone()) {
```

Comment:
Minor: this sequence can be simplified somewhat by reorganizing it into something like:

```
TaskStatus taskStatus = getTaskStatus();
if (taskStatus.getState().isDone()) {
  log.warn(...);
}
else {
  List<ExecutionFailureInfo> failures = ...;
  taskStatus = failWith(taskStatus, FAILED, failures);
}
updateTaskInfo(getTaskInfo().withTaskStatus(taskStatus));
```